### PR TITLE
Automatic download and installation of a release on instance start

### PR DESCRIPTION
**Pull request by aschaeffer**
If

* an instance shall be started
* and no suitable release is available

Then

1. download the best suitable release
2. install the downloaded release
3. start the instance using the installed release

### CODE REVIEW

File: src/instances/Instances.js [hunk 1]
```diff
@@ -263,58 +263,63 @@ class InstanceManager extends EventEmitter {
 
             // Resolve executable
             try {
-                const releaseNode = this.releaseManager.getRelease(instanceNode.versionrange, instanceNode.channel, true);
-                if (!releaseNode) {
+                this.releaseManager.getOrInstallRelease(instanceNode.versionrange, instanceNode.channel).then((releaseNode) => {
+                    this.log.debug(`Starting instance using release ${releaseNode.version} @ ${releaseNode.channel}`);
+                    this.log.debug('1');
+                    const executable_folder = this.releaseManager.getBinaryPath(releaseNode.version, releaseNode.channel);
+                    this.log.debug('2');
```
Comment (a-teammate) [inline src/instances/Instances.js]: maybe remove these 1-2-3 messages again?

File: src/releases/ReleaseManager.js [hunk 2]
```diff
@@ -546,14 +550,67 @@ class ReleaseManager extends EventEmitter {
...
+                    this.log.trace(`Found a release which is available, but not installed: ${availableReleaseNode.version} @ ${availableReleaseNode.channel}`);
+                    const version = availableReleaseNode.getChild('version').get();
+                    const channel = availableReleaseNode.getChild('channel').get();
+                    // Download release by release node
+                    this.downloadRelease(version, channel, false);
+//                    const onReleaseDownloadedHandler = () => {
+//                        // Install release by release node
+//                        this.log.trace(`Downloaded release ${version} @ ${channel}. Starting installation...`);
+//                        this.installRelease(version, channel);
+//                        this.once('onReleaseInstalled', () => {
```
Comment (a-teammate) [inline src/releases/ReleaseManager.js]: this can be removed I guess

File: src/instances/Instances.js (hunk 1 above)
Comment (aschaeffer) [reply src/instances/Instances.js]: lol, sure :-)

Comment (Croydon) [reply src/instances/Instances.js]: The good old log debugging 💓

**Pull request merged**